tests/functional/object/test_object_browser_client_provider_ui.py

```
@@ -342,161 +263,371 @@ def test_object_browser_list_objects_with_folders(self, project_factory):
+                    "Folder '%s' not found after upload" % folder_name
+                )
+                logger.info("Folder uploaded successfully")
+
+                # Step 3: Download object and verify integrity via S3
+                logger.test_step("Step 3: Downloading object and verifying integrity")
+                download_path = os.path.join(
+                    tempfile.gettempdir(), "downloaded_file1.txt"
+                )
+                obc["s3_client"].download_file(
+                    obc["bucket_name"],
+                    "%s/file1.txt" % folder_name,
```

Ruff (0.16.0)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)